packages/toolpad-app/src/server/FunctionsManager.ts

```diff
@@ -19,16 +19,20 @@ import { Awaitable } from '../utils/types';
 import { format } from '../utils/prettier';
 import { compilerOptions } from './functionsShared';
 
-function createDefaultFunction(): string {
-  return format(`
+async function createDefaultFunction(filePath: string) {
```

Comment:
To keep the return type for async functions you can always use `: Promise<string>`
```suggestion
async function createDefaultFunction(filePath: string): Promise<string> {
```

Reply:
I done